### ISSUE TITLE
Multiple Training Modality

### ISSUE THREAD

**Issue by mathnathan** [emelex-ai/BRIDGE#44]
## **Feature Request**

We need the capability to specify various training modalities. We begin by defining the desired training modalities:

- o -> p: Take the orthography as the input and the phonology as the targets
- p -> o: Take the phonology as the input and the orthography as the targets
- op -> op: Take both orthography and phonology as inputs and map to both orthography and phonology 

Endowing the model with the capacity to train under these three paradigms admits various ways to constrain the global embedding space. For example, we could alternate between o -> p and p -> o. This way, the global embedding space is forced to learn something useful to both pathways simultaneously. Or we could pretrain one pathway, say p -> o, and then train the join pathway op -> op and see how the model transfers what was learned in the p -> 0 training process.

### Implementation Details

The plan for implementing this involves X steps:

1. Add a new flag `pathway` to the `main.py` routine
2. Create three separate functions in `model.py` 
  - `o2p`: Uses only the internal parameters necessary for activating the orthography to phonology pathway
  - `p2o`: Uses only the internal parameters necessary for activating the phonology to orthography pathway
  - `op2op`: Uses all of the parameters, including the cross attention mixing layers, to the entire model
3. Call the three functions from the forward function depending on the flag specified
4. Modify the loss calculation within the forward pass depending on the flag specified

### Tests

@erlebach perhaps this should be in a separate issue, but once these new pathways have been implemented, we should devise a set of tests to ensure all of the operate correctly.
